enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,725 @@
+  * `reason` (optional, [string][json-string]), with a CamelCase reason, as described in [the blocked-edges section](#enhanced-graph-data-schema-for-blocking-edges).
+  * `message` (optional, [string][json-string]), with a human-oriented message describing the blocking reason, as described in [the blocked-edges section](#enhanced-graph-data-schema-for-blocking-edges).
+  * `blockedClusters` (optional, [object][json-object]).
+    If any `blockedClusters` property matches a given cluster, the edge is not recommended for that cluster, as described in [the blocked-edges section](#enhanced-graph-data-schema-for-blocking-edges).
+    If `blockedClusters` is unset, the edge is blocked for all clusters.
+    * `promql` (optional, [string][json-string]), defining the subset of affected clusters, as described in [the blocked-edges section](#enhanced-graph-data-schema-for-blocking-edges).
+
+### Enhanced ClusterVersion representation
+
+[The ClusterVersion `status`][api-cluster-version-status] will be extended with a new `conditionalUpdates` property:
+
+```go
+// conditionalUpdates contains the list of updates that may be
+// conditionally recommended for this cluster. Consumers interested
+// in the set of updates that are actually recommended for this
+// cluster should use availableUpdates. This list may be empty if no
+// updates are recommended, if the update service is unavailable, or
+// if an empty or invalid channel has been specified.
+// +optional
+conditionalUpdates []ConditionalUpdate `json:"conditionalUpdates,omitempty"`
+```
+
+The `availableUpdates` documentation will be adjusted to read:
+
+```go
+// availableUpdates contains the subset of conditionalUpdates that
+// apply to this cluster.  Updates which appear in conditionalUpdates
+// but not in availableUpdates may expose this cluster to known
+// issues.
+```
+
+The new ConditionalUpdate type will have the following schema:
+
+```go
+// ConditionalUpdate represents an update which is recommended to some
+// clusters on the version the current cluster is reconciling, but which
+// may not be recommended for the current cluster.
+// +k8s:deepcopy-gen=true
+type ConditionalUpdate struct {
+	// release is the target of the update.
+	// +required
+	release Release `json:"release"`
+
+	// blockedClusters represents the subset of clusters for which the conditional update is not recommended.
+	// +required
+	BlockedClusters []ConditionalUpdateBlocker `json:"blockedClusters"`
+
+	// conditions represents the observations of the conditional update's
+	// current status. Known types are:
+	// * Recommended, for whether the update is recommended for the current cluster.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
```

Comment:
RiskAssessment?

Reply:
Are the pros and cons of using `conditions` vs. dedicated properties laid out somewhere?  I liked it here because it sounded status-y, and helped distinguish between the CVO-inputs slurped out of the Cincinnati JSON and the CVO's own take on those inputs.  But I'd be fine using `recommended` or some such instead.  Or even `status`, if we feel fairly confident that the only thing the CVO will want to express about these conditional update targets is their recommended-ness.

Reply:
i'm also not clear on @deads2k point about multiplicity + oc wait.  can you clarify?

fundamentally i don't have a strong or informed opinion here, just trying to help @wking move forward.

Reply:
I've pushed 0305150 -> af6d5bb, keeping `conditions` and adding a new `Evaluating` condition, to allow folks to distinguish "CVO is working on it" from "CVO won't even try".  You could get at similar information implicitly by looking at whether or not the CVO had pushed a `Recommended` condition, but the `Evaluating` condition makes this explicit.

I think that addresses this thread, so I'll mark it resolved, but feel free to re-open if I'm misunderstanding the ask.